src/macros.jl

```diff
@@ -283,6 +283,101 @@ macro byrow(args...)
 end
 
 
+struct SpreadMissings{F} <: Function
```

Comment:
what is the rationale behind `SpreadMissings` name? It would be more natural for me to call it something like "complete cases" (or just `SkipMissings`).

Reply:
It was originally named for the "spreading" that happens when the shorter vector with the non-missing indices is spread out to match the length of the inputs. But I agree that "complete cases" is better.

Reply:
The problem with the name `SkipMissings` is that this type is quite different from `SkipMissing` due to the "spreading" behavior. Actually, this function can be defined as a vectorized `passmissing` when the wrapped function returns a vector; but when the function returns a scalar it leaves it as a scalar. This behavior is what seems to make the most sense for users: an alternative would be to "spread" scalars so that `missing` is used in positions which are `missing` in the input, but that doesn't sound super useful.

So it's hard to find a good name due to this hybrid behavior. I'd almost call it `ArrayPassMissing`. Anyway for now it's internal.

But what matters is the API that DataFramesMeta (and probably DataFrames at some point) exposes to users. The approach adopted here is to make `@passmissing` the go-to solution to handle missing values, which would work for `mean(:x)`, `cor(:x, :y)`, `scale(:x)`, etc. It's appealing as most users wouldn't have to care about the subtle differences between `passmissing` and `skipmissing` and their potential variants (`mean(skipmissing(:x))` would be an alternative to `@passmissing mean(:x)` which makes a single pass over the data, but most users probably don't care a lot).

Another approach would be to require users to choose what they want to do, but it would be much more annoying to use:
- `@skipmissing` would use `skipmissing∘f` for single arguments, or a new variant of `skipmissing` for multiple arguments which passes a view of their non-missing entries; both would leave the returned value as-is
- `@passmissing` would use `passmissing(f)` for `AbstractArray` inputs, or `spreadmissing(f)` for `AbstractArray` inputs, and would require the output to be an `AbstractArray` with its length equal to the number of non-missing entries in the input (or possibly broadcast a scalar result to non-missing positions)